app/src/main/java/com/hover/stax/transfers/TransferActivity.java

```diff
@@ -66,16 +66,31 @@ private void createFromSchedule(int schedule_id) {
 	}
 
 	private void createFromRequest(String link) {
-		AlertDialog dialog = new StaxDialog(this).setDialogMessage(R.string.loading_link_dialoghead).showIt();
 		transferViewModel.decrypt(link);
+		observeRequest();
+		observeActiveChannel();
+		Amplitude.getInstance().logEvent(getString(R.string.clicked_request_link));
+	}
+
+	private void observeRequest() {
+		AlertDialog dialog = new StaxDialog(this).setDialogMessage(R.string.loading_link_dialoghead).showIt();
 		transferViewModel.getRequest().observe(this, request -> {
 			Log.e(TAG, "maybe viewing request");
 			if (request == null) return;
+
 			Log.e(TAG, "viewing request " + request);
-			if (dialog != null) dialog.dismiss();
+			if (dialog != null) {
+				transferViewModel.setRecipientSmartly(request, channelDropdownViewModel.getActiveChannel().getValue());
```

Comment:
setRecipientSmartly should be called from the fragment. There is already an observer there for request, so this should go in the load function which is called when the request is loaded from the DB. Same to the channel observer, there is already one there, use it instead.